flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SchedulerTestingUtils.java

```diff
@@ -0,0 +1,168 @@
+import org.apache.flink.runtime.io.network.partition.NoOpJobMasterPartitionTracker;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.jobgraph.tasks.CheckpointCoordinatorConfiguration;
+import org.apache.flink.runtime.jobgraph.tasks.JobCheckpointingSettings;
+import org.apache.flink.runtime.messages.checkpoint.AcknowledgeCheckpoint;
+import org.apache.flink.runtime.metrics.groups.UnregisteredMetricGroups;
+import org.apache.flink.runtime.rest.handler.legacy.backpressure.VoidBackPressureStatsTracker;
+import org.apache.flink.runtime.scheduler.strategy.EagerSchedulingStrategy;
+import org.apache.flink.runtime.shuffle.NettyShuffleMaster;
+import org.apache.flink.runtime.taskmanager.TaskExecutionState;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * A utility class to create {@link DefaultScheduler} instances for testing.
+ */
+public class SchedulerTestingUtils {
+
+	private static final Logger LOG = LoggerFactory.getLogger(SchedulerTestingUtils.class);
+
+	public static DefaultScheduler createScheduler(
+			JobGraph jobGraph,
+			ManuallyTriggeredScheduledExecutorService asyncExecutor) throws Exception {
+
+		return new DefaultScheduler(
+			LOG,
+			jobGraph,
+			VoidBackPressureStatsTracker.INSTANCE,
+			Executors.directExecutor(),
+			new Configuration(),
+			new SimpleSlotProvider(jobGraph.getJobID(), 0),
+			asyncExecutor,
+			asyncExecutor,
+			ClassLoader.getSystemClassLoader(),
+			new StandaloneCheckpointRecoveryFactory(),
+			Time.seconds(300),
+			VoidBlobWriter.getInstance(),
+			UnregisteredMetricGroups.createUnregisteredJobManagerJobMetricGroup(),
+			Time.seconds(300),
+			NettyShuffleMaster.INSTANCE,
+			NoOpJobMasterPartitionTracker.INSTANCE,
+			new EagerSchedulingStrategy.Factory(),
+			new RestartPipelinedRegionStrategy.Factory(),
+			new TestRestartBackoffTimeStrategy(true, 0),
+			new DefaultExecutionVertexOperations(),
+			new ExecutionVertexVersioner(),
+			new TestExecutionSlotAllocatorFactory());
+	}
+
+	public static void enableCheckpointing(final JobGraph jobGraph) {
+		final List<JobVertexID> triggerVertices = new ArrayList<>();
+		final List<JobVertexID> allVertices = new ArrayList<>();
+
+		for (JobVertex vertex : jobGraph.getVertices()) {
+			if (vertex.isInputVertex()) {
+				triggerVertices.add(vertex.getID());
+			}
+			allVertices.add(vertex.getID());
+		}
+
+		final CheckpointCoordinatorConfiguration config = new CheckpointCoordinatorConfiguration(
+			Long.MAX_VALUE, // disable periodical checkpointing
+			10 * 60 * 1000,
```

Comment:
Nit, would be good to avoid magic values.

Reply:
What would you pass here instead?

Reply:
I was just thinking of a Macro.

Reply:
Will do